src/connection.cpp

```diff
@@ -54,7 +54,7 @@ void Connection::close(bool force)
 	connectionState = CONNECTION_STATE_DISCONNECTED;
 
 	if (protocol) {
-		g_dispatcher.addTask(createTask(std::bind(&Protocol::release, protocol)));
+		g_dispatcher.addTask(createTask([protocol = protocol]() { protocol->release(); }));
```

Comment:
It's strange but none of those ways of forming the lambda work for me
I am using Visual Studio 2019 on Windows 10 with an i7 processor.

for now `[protocol = protocol]` is the only way that doesn't cause the interpreter to crash because of the bad_weak_ptr exeption possibly this does not happen with more modern computers but I can not test it

Reply:
Yep, originally I was using `=`, on my compiler/OS it does not crash but it was crashing for @MillhioreBT

It is copied because the connection may be disposed before the protocol is released

Reply:
I see, strange.